### PR TITLE
Frontend changes for OT registration approvals

### PR DESCRIPTION
**Pull request by DanielRyanSmith**
This PR adds new fields to the OT creation request UI to facilitate the registration approvals settings for new origin trials.

The new fields are only available by selecting the checkbox for "Trial participation requires approval".

https://github.com/GoogleChrome/chromium-dashboard/assets/56164590/7fcfbd16-3e7a-4eb1-8917-3bec1cef4673

### CODE REVIEW

File: client-src/elements/form-field-specs.js [hunk 1]
```diff
@@ -1128,6 +1128,50 @@ export const ALL_FIELDS = {
...
   },
 
+  'ot_require_approvals': {
+    type: 'checkbox',
+    initial: false,
+    label: 'Trial participation requires approval',
+    help_text: html`
+      Will this trial require registrants to receive approval before
+      participating? See information about this setting at
+      <a target="_blank" href="">Some link</a>`,
```
Comment (past) [inline client-src/elements/form-field-specs.js:1138]: I don't think we want to provide a link here, but rather instruct feature authors to reach out to origin-trials-support@google.com to discuss their needs. Maybe replace "See information about this setting at Some link" with "Please reach out to origin-trials-support@google.com beforehand to discuss options here."

**Review by past — CHANGES_REQUESTED** [review]
LGTM other than the comment below.

File: client-src/elements/form-field-specs.js (hunk 1 above)
Comment (DanielRyanSmith) [reply client-src/elements/form-field-specs.js]: Description updated to:

"Will this trial require registrants to receive approval before participating? Please reach out to origin-trials-support@google.com beforehand to discuss options here."

File: client-src/elements/form-field-specs.js [hunk 2]
```diff
@@ -1128,6 +1128,55 @@ export const ALL_FIELDS = {
...
+  },
+
+  'ot_approval_criteria_url': {
+    type: 'input',
+    attrs: URL_FIELD_ATTRS,
+    required: true,
+    label: 'Approval criteria link',
+    help_text: html`
+    <p>
+      Link to documentation public describing the requirements
```
Comment (jrobbins) [inline client-src/elements/form-field-specs.js:1175]: "documentation public" -> "public documentation"

**Review by jrobbins — APPROVED** [review]
lgtm after making one wording fix

**Pull request merged**